src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/jpa/entity/AuthUser.kt

```diff
@@ -9,4 +9,16 @@ data class AuthUser(
   @Id @NotNull val id: String,
   @NotNull val authSource: String,
   @NotNull val userName: String,
-)
+) {
+  override fun hashCode(): Int {
+    return id.hashCode()
+  }
+
+  override fun equals(other: Any?): Boolean {
+    if (other == null || other !is AuthUser) {
+      return false
+    }
+
+    return id == other.id
```

Comment:
Whilst it will never be the case that ids match across sources, It seems that it's good form to include authSource in the equality as well.

Reply:
since the id is the db primary key i don't think it's nessecary

Reply:
i did initially include it. but later removed it.